src/util/file.cpp

```diff
@@ -86,3 +86,20 @@ QString getFilePathWithVerifiedExtensionFromFileDialog(
     }
     return fileLocation;
 }
+
+bool hasValidFilePathSyntax(const QString& path) {
+    QRegularExpression regex;
+#ifdef Q_OS_WIN
+    // Windows path with drive letter (C:\folder\file.txt) or UNC (\\server\folder\file.txt)
+    // or environment variable (%USERPROFILE%\folder\file.txt)
+    // or path with forward slash (C:/folder/file.txt)
+    // or relative path (folder/file.txt or ..\folder\file.txt)
+    regex.setPattern(
+            R"(^(?:[a-zA-Z]:\\|\\\\[\w.]+\\[\w.$]+|%\w+%\\|.*\/.*|.*\\.*|[^\\/:*?"<>|\r\n]+$))");
+#else
+    // Unix path (/folder/file.txt) or with environment variable ($HOME/folder/file.txt)
+    // or relative path (folder/file.txt or ../folder/file.txt)
+    regex.setPattern(R"(^(/|\$[\w]+/|.*\/.*|[^/]*[^/:*?"<>|\r\n]+$))");
+#endif
+    return regex.match(path).hasMatch();
```

Comment:
It's a different functionality. We need to check here if the path syntax is valid, not if the particular file exists.

Reply:
if we could only check for validity I would've suggested that, but QFileInfo doesn't have that feature afaict. Checking for the existence also checks for validity. Do you see a real usecase where the CSVParser above would accept a valid path that doesn't exist?

Reply:
Yes, the parser just checks the first row for valid path syntax to detect the location column. If we would change this to file existence and this row of the playlist would contain a path to a non existing file, the whole playlist would become unreadable.

Reply:
I see, but that sounds more like a flaw in the parser (failing to validate a single entry causing failure to read the entire playlist).
Instead of people being tempted to rely on this function when its in `util/file.h` can you move it to the parser and label it as a heuristic more clearly?